src/simsopt/geo/curve.py

```diff
@@ -841,14 +937,227 @@ def wrap(data):
 
     polyLinesToVTK(str(filename), x, y, z, pointsPerLine=ppl, pointData=pointData)
 
 
-def create_equally_spaced_curves(ncurves, nfp, stellsym, R0=1.0, R1=0.5, order=6, numquadpoints=None):
+def setup_uniform_grid(s, s_inner, s_outer, Nx, Ny, Nz, Nmin_factor=2.5):
+    """
+    Generate a uniform 3D grid of points between two toroidal surfaces, with spacing and filtering to avoid coil 
+    overlap and respect stellarator symmetry.
+
+    This function is typically used to initialize candidate coil center locations for planar coil optimization. 
+    It computes a uniform grid in the bounding box between two surfaces, then (for nfp > 1) removes points 
+    outside the fundamental sector and those too close to the symmetry plane, to avoid overlap after symmetry operations.
+
+    Parameters
+    ----------
+    s : Surface
+        The main surface object (used for nfp and geometry info).
+    s_inner : Surface
+        The inner toroidal surface (for grid bounding box).
+    s_outer : Surface
+        The outer toroidal surface (for grid bounding box).
+    Nx : int
+        Number of grid points in the x direction.
+    Ny : int
+        Number of grid points in the y direction.
+    Nz : int
+        Number of grid points in the z direction.
+    Nmin_factor : float, optional
+        Factor to set minimum coil spacing (default: 2.5). The coil radius is set to Nmin / Nmin_factor, where Nmin is the minimum grid spacing.
+
+    Returns
+    -------
+    xyz_uniform : ndarray, shape (N, 3)
+        Array of candidate coil center points in 3D, filtered for symmetry and spacing.
+    xyz_inner : ndarray, shape (M, 3)
+        Array of points on the inner surface.
+    xyz_outer : ndarray, shape (M, 3)
+        Array of points on the outer surface.
+    R : float
+        The coil radius used for spacing.
+
+    Notes
+    -----
+    - For nfp > 1, points outside the fundamental sector are removed, and points too close to the symmetry plane 
+      are also filtered out to avoid overlap after symmetry operations.
+    - The returned grid is suitable for initializing planar coil centers in optimization routines.
+    """
+    # Get (X, Y, Z) coordinates of the two boundaries
+    nfp = s.nfp
+    xyz_inner = s_inner.gamma().reshape(-1, 3)
+    xyz_outer = s_outer.gamma().reshape(-1, 3)
+    x_outer = xyz_outer[:, 0]
+    y_outer = xyz_outer[:, 1]
+    z_outer = xyz_outer[:, 2]
+    x_max = np.max(x_outer)
+    x_min = np.min(x_outer)
+    y_max = np.max(y_outer)
+    y_min = np.min(y_outer)
+    z_max = np.max(z_outer)
+    z_min = np.min(z_outer)
+    z_max = max(z_max, abs(z_min))
+
+    # Initialize uniform grid
+    dx = (x_max - x_min) / (Nx - 1)
+    dy = (y_max - y_min) / (Ny - 1)
+    dz = 2 * z_max / (Nz - 1)
+    Nmin = min(dx, min(dy, dz))
+
+    # This is not a guarantee that coils will not touch but inductance
+    # matrix blows up if they do so it is easy to tell when they do
+    R = Nmin / Nmin_factor
```

Comment:
why doesn't this guarantee the coils won't overlap? if R < h/2 isn't that enough?